src/js/components/action.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import StripeCheckout from 'react-stripe-checkout';
+
+import { store } from 'stores/app-store';
+import { createTransaction } from 'services/stripe-service';
+import { immediateUpdate } from 'services/user-service';
+
+import { LoadingComponent } from 'components/loading';
+
+export class ActionComponent extends Component {
+    constructor(...args) {
+        super(...args);
+
+        this.state = {
+            state: this.props.state
+        };
+    }
+
+    onStripeToken(token) {
+        let user = store.getState().user;
+        let promises = [];
+        if (!user.email) {
+            promises.push(immediateUpdate({
+                email: token.email
+            }));
+        }
+
+        let transactionPromise = createTransaction(this.props._id, token.id).then(() => {
+            this.setState({
```

Comment:
Does this get translated into some change in the store? If the component needs to be redrawn and loses state, can it regain this state from `store.conversation.messages.actions`, or wherever it's persisted?